src/main/java/xades4j/providers/impl/HttpTimeStampTokenProvider.java

```diff
@@ -148,27 +55,41 @@ private InputStream getResponse(byte[] encodedRequest) throws TimeStampTokenGene
             out.write(encodedRequest);
             out.flush();
 
-            if (connection.getResponseCode() != HttpURLConnection.HTTP_OK)
-            {
+            if (connection.getResponseCode() != HttpURLConnection.HTTP_OK) {
                 throw new TimeStampTokenGenerationException(String.format("TSA returned HTTP %d %s", connection.getResponseCode(), connection.getResponseMessage()));
             }
 
-            // TODO do we need to invoke connection.disconnect()?
-            return new BufferedInputStream(connection.getInputStream());
-        } catch (IOException ex)
-        {
+            try (BufferedInputStream input = new BufferedInputStream(connection.getInputStream())) {
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
Why is this better than returning the stream, use it on the constructor of `TimeStampResponse` and close it afterwards, all in the base class? This way the bytes/stream are iterated twice.